ReactAndroid/src/main/java/com/facebook/react/shell/MainPackageConfig.java

```diff
@@ -10,31 +10,51 @@
 package com.facebook.react.shell;
 
 import com.facebook.imagepipeline.core.ImagePipelineConfig;
+import com.facebook.react.modules.network.DefaultOkHttpProvider;
+import com.facebook.react.modules.network.OkHttpClientProvider;
 
 /**
  * Configuration for {@link MainReactPackage}
  */
 public class MainPackageConfig {
 
-  private ImagePipelineConfig mFrescoConfig;
+
+  private final OkHttpClientProvider mOkHttpClientProvider;
+
+  private final ImagePipelineConfig mFrescoConfig;
 
   private MainPackageConfig(Builder builder) {
     mFrescoConfig = builder.mFrescoConfig;
+    mOkHttpClientProvider = builder.mOkHttpClientProvider;
   }
 
   public ImagePipelineConfig getFrescoConfig() {
     return mFrescoConfig;
   }
 
+  public OkHttpClientProvider getOkHttpClientProvider() {
+    return mOkHttpClientProvider != null ? mOkHttpClientProvider : new DefaultOkHttpProvider();
+  }
+
   public static class Builder {
 
     private ImagePipelineConfig mFrescoConfig;
+    private OkHttpClientProvider mOkHttpClientProvider;
 
     public Builder setFrescoConfig(ImagePipelineConfig frescoConfig) {
       mFrescoConfig = frescoConfig;
       return this;
     }
 
+    /**
+     * Ability to inject configured OkHttpClient provider to react-native networking subsystem.
```

Comment:
Same comment about the {@link} annotation. I also dont think that the sentence `This will be very useful` is needed.